frontend/src/features/queue/hooks/hooks.ts

```diff
@@ -0,0 +1,264 @@
+import { useContext } from "react";
+import OnRouteBCContext from "../../../common/authentication/OnRouteBCContext";
+import { IDIRUserRoleType } from "../../../common/authentication/types";
+import { isStaffUser } from "../helpers/isStaffUser";
+import { MRT_PaginationState, MRT_SortingState } from "material-react-table";
+import {
+  keepPreviousData,
+  useMutation,
+  useQuery,
+  useQueryClient,
+} from "@tanstack/react-query";
+import {
+  claimApplicationInQueue,
+  getApplicationInQueueDetails,
+  getApplicationsInQueue,
+  getClaimedApplicationsInQueue,
+  getUnclaimedApplicationsInQueue,
+  updateApplicationQueueStatus,
+} from "../apiManager/queueAPI";
+import { Nullable } from "../../../common/types/common";
+import { SnackBarContext } from "../../../App";
+import { CASE_ACTIVITY_TYPES } from "../types/CaseActivityType";
+import { AxiosError } from "axios";
+import { useTableControls } from "../../permits/hooks/useTableControls";
+
+const QUEUE_QUERY_KEYS = {
+  BASE: () => ["queue"],
+  ALL: (pagination: MRT_PaginationState, sorting: MRT_SortingState) => [
+    [...QUEUE_QUERY_KEYS.BASE(), pagination, sorting],
```

Comment:
You could just create a const like ```const QUEUE_QUERY_KEY_BASE = "queue";``` above, and reused it here as well as below as ```[QUEUE_QUERY_KEY_BASE, pagination, sorting]```

Reply:
It looks less awkward (without the unnecessary destructuring)